apps/experiments/models.py

```diff
@@ -974,6 +974,28 @@ def format_custom_action_operation(op) -> str:
             ],
         )
 
+    def get_assistant(self):
+        """
+        Retrieves the assistant associated with the current instance.
+
+        This method attempts to find an assistant node within the pipeline associated with the current instance.
+        - If an assistant node is found, it retrieves the assistant ID from the node's parameters and returns the
+        corresponding OpenAiAssistant object.
+        - If no assistant node is found or if the pipeline is not set, it returns the default assistant associated with
+        the instance.
+        """
+        from apps.assistants.models import OpenAiAssistant
+        from apps.pipelines.models import Node
+        from apps.pipelines.nodes.nodes import AssistantNode
+
+        if self.pipeline:
+            node_name = AssistantNode.model_json_schema()["title"]
+            # TODO: What about multiple assistant nodes?
+            assistant_node = Node.objects.filter(type=node_name, pipeline=self.pipeline).first()
+            if assistant_node and assistant_node.params.get("assistant_id"):
```

Comment:
nit: I think you could add this in the filter:

```
Node.objects.filter(..., params__assistant_id__isnull=False).values_list("params__assistant_id", flat=True).first()
```

Reply:
Good call, I'll update

Reply:
https://github.com/dimagi/open-chat-studio/pull/869/commits/d166b39ff88e83eedd3aa7f38909bf98417781e6